flink-dist/src/main/flink-bin/bin/flink

```diff
@@ -17,20 +17,41 @@
 # limitations under the License.
 ################################################################################
 
-target="$0"
 # For the case, the executable has been directly symlinked, figure out
 # the correct bin path by following its symlink up to an upper bound.
 # Note: we can't use the readlink utility here if we want to be POSIX
 # compatible.
-iteration=0
-while [ -L "$target" ]; do
-    if [ "$iteration" -gt 100 ]; then
-        echo "Cannot resolve path: You have a cyclic symlink in $target."
+followSymLink() {
+    local iteration=0
+    local bar=$1
+    while [ -L "$bar" ]; do
```

Comment:
+1 for local variables but please don't call them `bar` 😄 How about `local target=$1`?